libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,329 @@
+more information about importing and exporting your modified objects, see
+{kibana-ref}/managing-saved-objects.html[Managing Saved Objects].
+
+To re-import the dashboards, run the `setup` command with the `--dashboards`
+option specified. For example:
 
 [source,shell]
-------------------------------------------------------------------------------
-metricbeat-6.0.0-2017-08-31
-------------------------------------------------------------------------------
+----
+metricbeat setup --dashboards
+----
 
-And the corresponding Elasticsearch template is named `metricbeat-6.0.0`.
+Also, if you plan to run {beats} 6.7 and 7.0 in parallel,
+<<enable-ecs-compatibility,enable the compatibility layer>> *before* you start
+ingesting data. 
 
-This means that each version of the Beat creates a new index, and it is
-guaranteed that the correct template for that version is applied.  With these
-changes in place, you generally don't have to do anything to upgrade the mapping
-template when you move to a new version.
+[[migrate-filebeat-registry]]
+==== Migrate {filebeat} registry to use new format
+
+Starting with version 7.0, {filebeat} stores the registry in a directory
+structure. If Filebeat finds an old registry file at the path set by
+`filebeat.registry.path`, it will automatically migrate the registry file to the
+new format. If the registry file is at a different location, set
+`filebeat.registry.migrate_file` to point to the file.
+
+The registry changes also require you to rename the following configuration
+settings:
+
+[options="header"]
+|====
+|Old config name | New config name
+|`filebeat.registry_file`    | `filebeat.registry.path`.
+|`registry_file_permissions` | `registry.file_permissions`
+|`registry_flush` | `registry.flush`
+|====
+
+Before proceeding with the upgrade, make sure you back up the registry file.
+
+[role="xpack"]
+[[ilm-on]]
+==== Check privileges for index lifecycle management (on by default in 7.0)
+
+Staring with {beats} 7.0, index lifecycle management is on by default when
+sending data to {beats} clusters that support it. Make sure {beats} users have
+the privileges needed to use index lifecycle management, or disable index
+lifecycle management. 
+
+For help troubleshooting authorization issues, see <<user-unauthorized>>.
+
+If you want to disable index lifecycle management, set
+`setup.ilm.enabled: false` in the {beats} configuration file.
+
+[[troubleshooting-upgrade]]
+=== Troubleshooting {beats} upgrade issues
+
+This section describes common problems you might encounter when upgrading to
+{beats} 7.x.
+
+You can avoid some of these problems by reading <<upgrading-6-to-7>> before
+upgrading {beats}.
+
+[[missing-fields]]
+==== Missing fields or too many fields in the index
+
+You may have run the Beat before loading the required index template. To clean
+up and start again:
+
+. Delete the index that was created when you ran the Beat. For example:
```

Comment:
I guess there will be quite a few cases where users started indexing, have the wrong format, then load the index template and wonder why they still have the wrong format. So they need to delete index template and indices.

Reply:
added a step about deleting the template, too